src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1983,7 +2334,7 @@ public override BoundNode VisitNullCoalescingAssignmentOperator(BoundNullCoalesc
             TrackNullableStateForAssignment(rightOperand, targetType, leftSlot, rightResult, MakeSlot(rightOperand));
             Join(ref this.State, ref leftState);
 
-            TypeWithState resultType = GetNullCoalescingResultType(leftOperand, currentLeftType, rightOperand, rightResult, targetType.TypeSymbol);
+            TypeWithState resultType = GetNullCoalescingResultType(leftOperand, currentLeftType, rightOperand, rightResult, node.Type);
```

Comment:
>node.Type [](start = 124, length = 9)

This change seems wrong. Types from binding are generally ignored once we have a result type from visiting a node.